File: frontend_flutter/lib/modules/sales/domain/entities/order_summary.dart
/// Domain entity. No Flutter, no JSON.
/// Bảng Tổng hợp Đơn đặt hàng: read-only, scoped by sale admin (owner).
class OrderSummaryItem {
  const OrderSummaryItem({
    required this.productCode,
    required this.productName,
    required this.packageSize,
    required this.packageUnit,
    required this.productType,
    required this.packagingType,
    required this.quantity,
  });

  final String productCode;
  final String productName;
  final String packageSize;
  final String packageUnit;
  final String productType;
  final String packagingType;
  final int quantity;

  /// Quy cách = packageSize + packageUnit (e.g. 100gr).
  String get specification => '$packageSize$packageUnit';

  /// Dạng = productType; Đóng gói = packagingType.
}

/// Detail view (with items). From GET by-date or GET by-id.
class OrderSummary {
  const OrderSummary({
    required this.id,
    required this.ownerId,
    required this.summaryDate,
    required this.createdAt,
    this.approvedBy,
    required this.items,
  });

  final String id;
  final String ownerId;
  final String summaryDate; // ISO date or yyyy-MM-dd
  final DateTime createdAt;
  final String? approvedBy;
  final List<OrderSummaryItem> items;
}

/// List entry (no items). From GET order-summaries.
class OrderSummaryListEntry {
  const OrderSummaryListEntry({
    required this.id,
    required this.ownerId,
    required this.summaryDate,
    required this.createdAt,
    required this.itemCount,
  });

  final String id;
  final String ownerId;
  final String summaryDate;
  final DateTime createdAt;
  final int itemCount;
}

/// Paginated list result for order summaries.
class OrderSummaryListResult {
  const OrderSummaryListResult({
    required this.items,
    required this.total,
  });
  final List<OrderSummaryListEntry> items;
  final int total;
}

File: backend/internal/dto/order_summary.go
package dto

import (
	"github.com/google/uuid"
	"time"
)

// OrderSummaryItemPayload for create/update request.
type OrderSummaryItemPayload struct {
	ProductID uuid.UUID `json:"productId"`
	Quantity  int       `json:"quantity"`
}

// OrderSummaryItemDetail for detail response (includes product info).
type OrderSummaryItemDetail struct {
	ProductID       uuid.UUID `json:"productId"`
	ProductCode     string    `json:"productCode"`
	ProductName     string    `json:"productName"`
	PackageSize     string    `json:"packageSize"`
	PackageUnit     string    `json:"packageUnit"`
	ProductType     string    `json:"productType"`
	PackagingType   string    `json:"packagingType"`
	Quantity        int       `json:"quantity"`
}

// CreateOrderSummaryRequest for POST order-summaries.
type CreateOrderSummaryRequest struct {
	SummaryDate string                    `json:"summaryDate"` // YYYY-MM-DD (ngày tổng hợp đơn)
	Items       []OrderSummaryItemPayload `json:"items"`
}

// UpdateOrderSummaryRequest for PUT order-summaries/:id.
type UpdateOrderSummaryRequest struct {
	SummaryDate string                    `json:"summaryDate"` // YYYY-MM-DD; must remain unique per day
	Items       []OrderSummaryItemPayload `json:"items"`
}

// OrderSummaryPayload for list.
type OrderSummaryPayload struct {
	ID          uuid.UUID   `json:"id"`
	OwnerID     uuid.UUID   `json:"ownerId"` // sale admin who owns this summary
	SummaryDate time.Time   `json:"summaryDate"`
	CreatedAt   time.Time   `json:"createdAt"`
	CreatedBy   uuid.UUID   `json:"createdBy"`
	ApprovedBy  *uuid.UUID  `json:"approvedBy,omitempty"`
	ItemCount   int         `json:"itemCount"`
}

// OrderSummaryDetailPayload for get by id (with items).
type OrderSummaryDetailPayload struct {
	OrderSummaryPayload
	Items []OrderSummaryItemDetail `json:"items"`
}
